### DISCUSSION TITLE
Allow failures for remote secrets/remote state references

### DISCUSSION THREAD

**Discussion by andrewnazarov** [helmfile/helmfile#111] [Ideas]
### Preface
We have a common helmfile where we define all releases that must be deployed to all environments and then for each environment (dev, staging and production) we have their own help files which include that common helmfile as a subhelmfile (via `helmfiles:`).

We utilise `ref+` references in the common helmfile extensively. We ran `helmfile lint` to catch errors. However, since it's triggered for the default environment and this default environment doesn't have real things behind it (it doesn't have a terraform state to refer to, it doesn't have a corresponding GCP project with secrets, etc) a pipeline fails.

### Current workaround
We have to wrap ref+ references in `{{- if ne .Environment.Name "default" }}` constructions. It works, but it's far from perfect.
Ex.:
```
...
service:
  loadBalancerIP: {{- if ne .Environment.Name "default" }} ref+tfstategs://{{ env "PATH_TO_TF_STATE" }}/output.external_ip {{- end }}
```

### Suggestion
The first thing that comes to my mind is a flag that disables the `ref+` rendering. The default value might be another option, although I can't imagine what it might look like.

What do you think? Thanx!

**Comment by andrewnazarov** [comment]
Ok, what I found is this `fetchSecretValue` which can be used as a better approach. For example, for the default environment I can set all properties to some dummy values and for real envs - to real references. The only downside is that I have to repeat the keys in all environments values.

**Comment by balagurrr** [comment]
I think this is a useful feature.
